services/outputhost/ackmanager.go

```diff
@@ -65,12 +59,11 @@ type (
 		cgUUID             string
 		extUUID            string
 		connectedStoreUUID *string
-		*levels                    // ‡ the current levels
-		prev               *levels // ‡ the previous levels
+		*levels            // ‡ the current levels
 		ackLevelTicker     *time.Ticker
 		closeChannel       chan struct{}
 		waitConsumed       chan<- bool // waitConsumed is the channel which will signal if the extent is completely consumed given by extentCache
-		metaclient         metadata.TChanMetadataService
+		committer          Committer
```

Comment:
Just a general thought/question.. will it make sense to move the entire ackMgr data structure within the committer? i.e, the map which we keep track of.. this way we don't need the outer lock for the ackMgr and then a separate lock for the committer and we can use the committer to do everything.. of course this means the entire logic now moves to committer..